akka-actor-typed-tests/src/test/java/jdocs/akka/typed/AggregatorTest.java

```diff
@@ -0,0 +1,238 @@
+  }
+
+  interface IllustrateUsage {
+    // #usage
+    public class Hotel1 {
+      public static class RequestQuote {
+        public final ActorRef<Quote> replyTo;
+
+        public RequestQuote(ActorRef<Quote> replyTo) {
+          this.replyTo = replyTo;
+        }
+      }
+
+      public static class Quote {
+        public final String hotel;
+        public final BigDecimal price;
+
+        public Quote(String hotel, BigDecimal price) {
+          this.hotel = hotel;
+          this.price = price;
+        }
+      }
+    }
+
+    public class Hotel2 {
+      public static class RequestPrice {
+        public final ActorRef<Price> replyTo;
+
+        public RequestPrice(ActorRef<Price> replyTo) {
+          this.replyTo = replyTo;
+        }
+      }
+
+      public static class Price {
+        public final String hotel;
+        public final BigDecimal price;
+
+        public Price(String hotel, BigDecimal price) {
+          this.hotel = hotel;
+          this.price = price;
+        }
+      }
+    }
+
+    public class HotelCustomer extends AbstractBehavior<HotelCustomer.Command> {
+
+      interface Command {}
+
+      public static class Quote {
+        public final String hotel;
+        public final BigDecimal price;
+
+        public Quote(String hotel, BigDecimal price) {
+          this.hotel = hotel;
+          this.price = price;
+        }
+      }
+
+      public static class AggregatedQuotes implements Command {
+        public final List<Quote> quotes;
+
+        public AggregatedQuotes(List<Quote> quotes) {
+          this.quotes = quotes;
+        }
+      }
+
+      public static Behavior<Command> create(
+          ActorRef<Hotel1.RequestQuote> hotel1, ActorRef<Hotel2.RequestPrice> hotel2) {
+        return Behaviors.setup(context -> new HotelCustomer(context, hotel1, hotel2));
+      }
+
+      private final ActorContext<Command> context;
+
+      public HotelCustomer(
+          ActorContext<Command> context,
+          ActorRef<Hotel1.RequestQuote> hotel1,
+          ActorRef<Hotel2.RequestPrice> hotel2) {
+        this.context = context;
+
+        Consumer<ActorRef<Object>> sendRequests =
+            replyTo -> {
+              hotel1.tell(new Hotel1.RequestQuote(replyTo.narrow()));
+              hotel2.tell(new Hotel2.RequestPrice(replyTo.narrow()));
+            };
+
+        int expectedReplies = 2;
+        // Object since no common type between Hotel1 and Hotel2
+        context.spawnAnonymous(
+            Aggregator.create(
+                Object.class,
+                sendRequests,
+                expectedReplies,
+                context.getSelf(),
+                this::aggregateReplies,
+                Duration.ofSeconds(5)));
+      }
+
+      private AggregatedQuotes aggregateReplies(List<Object> replies) {
+        List<Quote> quotes =
+            replies
+                .stream()
+                .map(
+                    r -> {
```

Comment:
Maybe add a note here that we are adapting the different hotel protocols, just as extra hint, I think that could easily be overlooked.